libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -695,8 +697,9 @@ private void subscribeViewModels() {
     instructionView.subscribe(navigationViewModel);
     summaryBottomSheet.subscribe(navigationViewModel);
 
-    NavigationViewSubscriber subscriber = new NavigationViewSubscriber(navigationPresenter);
-    subscriber.subscribe(((LifecycleOwner) getContext()), navigationViewModel);
+    navigationViesSubscriber = new NavigationViewSubscriber(navigationPresenter);
+    lifecycleOwner = (LifecycleOwner) getContext();
```

Comment:
I think a more idiomatic way to handle the whole unsubscribe situation for you is to use a different `LifecycleOwner`. By tying it to `getContext()` you need these manual unsubscribe calls which is defeating the entire automation around using a `LifecycleOwner`. Views do not inherently have their own `LifecycleOwner` but that would be easy enough to create and keep a reference to. That would also avoid needing the modify the view model.

I am a little concerned `getContext()` is still being used as the `LifecycleOwner`. I would have to run / debug myself to eliminate those conerns

Reply:
that would involve adding something like this in `onCreate`

```
lifecycleRegistry = new LifecycleRegistry(this);
lifecycleRegistry.markState(Lifecycle.State.INITIALIZED);
```
and setting the various `markState(...)` in appropriate lifecycle methods